### PR TITLE
Edit product

### PR DESCRIPTION
**Pull request by Maha-ElMais**

### CODE REVIEW

File: app/controllers/products_controller.rb [hunk 1]
```diff
@@ -25,6 +25,37 @@ def create
...
+    driver_id = params[:id]
+
+    begin
+      @product = Product.find(driver_id)
+    rescue ActiveRecord::RecordNotFound
+      @product = nil
+    end
+
+    if @product.nil?
+      head :not_found
```
Comment (jwinchan) [inline app/controllers/products_controller.rb]: should use flash[:error] instead of head

File: app/controllers/products_controller.rb [hunk 2]
```diff
@@ -25,6 +25,37 @@ def create
   end
 
   def edit
+    @product = Product.find_by(id: params[:id])
+    driver_id = params[:id]
```
Comment (jwinchan) [inline app/controllers/products_controller.rb]: driver_id?

File: app/controllers/products_controller.rb [hunk 3]
```diff
@@ -25,6 +25,37 @@ def create
...
+    @product = Product.find_by(id: params[:id])
+    if @product.nil?
+      head :not_found
+      return
+    elsif @product.update(
+        name: params[:product][:name],
+        vin: params[:product][:vin],
+        available: params[:product][:available]
+    )
+      redirect_to driver_path # go to the product details page
```
Comment (jwinchan) [inline app/controllers/products_controller.rb]: use strong params

File: app/controllers/products_controller.rb [hunk 4]
```diff
@@ -25,6 +25,37 @@ def create
...
+    if @product.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def update
+    @product = Product.find_by(id: params[:id])
+    if @product.nil?
+      head :not_found
```
Comment (jwinchan) [inline app/controllers/products_controller.rb]: flash[:error]

File: app/controllers/products_controller.rb [hunk 5]
```diff
@@ -25,6 +25,37 @@ def create
...
+      return
+    elsif @product.update(
+        name: params[:product][:name],
+        vin: params[:product][:vin],
+        available: params[:product][:available]
+    )
+      redirect_to driver_path # go to the product details page
+      return
+    else # save failed
+    render :edit, status: :bad_request # show the new product form view again
```
Comment (jwinchan) [inline app/controllers/products_controller.rb:44]: insert flash[:error]

File: app/controllers/products_controller.rb [hunk 6]
```diff
@@ -25,6 +25,37 @@ def create
   end
 
   def edit
+    @product = Product.find_by(id: params[:id])
+    driver_id = params[:id]
+
+    begin
+      @product = Product.find(driver_id)
```
Comment (jwinchan) [inline app/controllers/products_controller.rb]: should be product_id

File: app/controllers/products_controller.rb (hunk 2 above)
Comment (Schmarj3) [reply app/controllers/products_controller.rb]: Also, can use Jessica's find_product method. So should be able to remove the @product = Product.find_by....

**Pull request merged**